### PR TITLE
Update BigQuery to use new Create/Update pattern

### PR DESCRIPTION
**Pull request by mccurdyc**
## Proposed Change(s)

* Uses a factory pattern for Creates and Updates.
* Adds test for non-exported API in `_test.go` file.
* Moves exported API test to `_integration_test.go` file.

## Validation

```
$ make test
```

```
git clone git@github.com:mccurdyc/cli.git /tmp/cli && \
(
    cd /tmp/cli && \
    git checkout mccurdyc/logging-test-bigquery && \
    make test && \
    rm -rf /tmp/cli \
)
```

### CODE REVIEW

File: pkg/logging/bigquery/bigquery_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,414 @@
+package bigquery_test
```
Comment (mccurdyc) [inline pkg/logging/bigquery/bigquery_integration_test.go:1]: `mv bigquery_test.go bigquery_integration_test.go`

File: pkg/logging/bigquery/create.go [hunk 2]
```diff
@@ -24,32 +37,71 @@ func NewCreateCommand(parent common.Registerer, globals *config.Data) *CreateCom
...
+	c.CmdClause.Flag("user", "Your Google Cloud Platform service account email address. The client_email field in your service account authentication JSON.").Required().StringVar(&c.User)
+	c.CmdClause.Flag("secret-key", "Your Google Cloud Platform account secret key. The private_key field in your service account authentication JSON.").Required().StringVar(&c.SecretKey)
+
+	c.CmdClause.Flag("template-suffix", "BigQuery table name suffix template").Action(c.Template.Set).StringVar(&c.Template.Value)
+	c.CmdClause.Flag("format", "Apache style log formatting. Must produce JSON that matches the schema of your BigQuery table").Action(c.Format.Set).StringVar(&c.Format.Value)
+	c.CmdClause.Flag("format-version", "The version of the custom logging format used for the configured endpoint. Can be either 2 (the default, version 2 log format) or 1 (the version 1 log format). The logging call gets placed by default in vcl_log if format_version is set to 2 and in vcl_deliver if format_version is set to 1").Action(c.FormatVersion.Set).UintVar(&c.FormatVersion.Value)
 	// TODO(phamann): It seems `placement` and `response_condition` aren't
```
Comment (mccurdyc) [inline pkg/logging/bigquery/create.go:54]: These are now exposed, but I will add them in a subsequent PR.

Comment (phamann) [reply pkg/logging/bigquery/create.go:54]: Great news! 🎉

**Comment by mccurdyc** [comment]
Force-pushed rebasing master.

**Pull request merged**